File: app/src/main/kt/estyle/teabaike/fragment/DrawerFragment.java
package estyle.teabaike.fragment;


import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.NavigationView;
import android.support.v4.app.Fragment;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import estyle.teabaike.R;
import estyle.teabaike.activity.CollectionActivity;
import estyle.teabaike.activity.CopyrightActivity;
import estyle.teabaike.activity.SearchActivity;
import estyle.teabaike.activity.SuggestionActivity;
import estyle.teabaike.databinding.FragmentDrawerBinding;

public class DrawerFragment extends Fragment implements
        NavigationView.OnNavigationItemSelectedListener, View.OnClickListener {

    private EditText mKeywordEditText;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        FragmentDrawerBinding binding = DataBindingUtil.inflate(inflater,
                R.layout.fragment_drawer,
                container, false);
        NavigationView rootView = (NavigationView) binding.getRoot();
        rootView.setNavigationItemSelectedListener(this);
        View headerView = rootView.getHeaderView(0);
        mKeywordEditText = headerView.findViewById(R.id.keyword_et);
        Button searchBtn = headerView.findViewById(R.id.search_btn);
        searchBtn.setOnClickListener(this);
        return rootView;
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.my_collection_item:
                CollectionActivity.startActivity(getContext());
                break;
            case R.id.copyright_info_item:
                CopyrightActivity.startActivity(getContext());
                break;
            case R.id.suggestion_item:
                SuggestionActivity.startActivity(getContext());
                break;
        }
        return true;
    }

    @Override
    public void onClick(View v) {
        String keyword = mKeywordEditText.getText().toString();
        if (!TextUtils.isEmpty(keyword)) {
            SearchActivity.startActivity(getContext(), keyword);
        }
    }
}

File: app/src/main/kt/estyle/teabaike/fragment/DrawerFragment.kt
package estyle.teabaike.fragment


import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import com.google.android.material.navigation.NavigationView
import estyle.base.BaseFragment
import estyle.teabaike.R
import estyle.teabaike.activity.CollectionActivity
import estyle.teabaike.activity.CopyrightActivity
import estyle.teabaike.activity.FeedbackActivity
import estyle.teabaike.activity.SearchActivity

class DrawerFragment : BaseFragment(), NavigationView.OnNavigationItemSelectedListener,
    View.OnClickListener {

    private var mKeywordEditText: EditText? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_drawer, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val rootView = view as NavigationView
        rootView.setNavigationItemSelectedListener(this)
        val headerView = rootView.getHeaderView(0)
        mKeywordEditText = headerView.findViewById(R.id.keyword_et)
        val searchBtn = headerView.findViewById<Button>(R.id.search_btn)
        searchBtn.setOnClickListener(this)
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.my_collection_item -> CollectionActivity.startActivity(context!!)
            R.id.copyright_info_item -> CopyrightActivity.startActivity(context!!)
            R.id.suggestion_item -> FeedbackActivity.startActivity(context!!)
        }
        return true
    }

    override fun onClick(v: View) {
        val keyword = mKeywordEditText!!.text.toString()
        if (!TextUtils.isEmpty(keyword)) {
            SearchActivity.startActivity(context!!, keyword)
        }
    }
}
